--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,49 +93,7 @@ set(MINIZIP_DEPENDS "" CACHE INTERNAL "MINIZIP_DEPENDS")
 
 set(LIBKML_TARGETS)
 
-find_package(EXPAT)
-if(EXPAT_FOUND)
-  include_directories(${EXPAT_INCLUDE_DIR})
-else()
-  include(External_expat)
-  list(APPEND KMLBASE_DEPENDS EXPAT)
-endif()
-
-find_package(ZLIB 1.2.8)
-if(ZLIB_FOUND)
-  include_directories(${ZLIB_INCLUDE_DIR})
-else()
-  include(External_zlib)
-  list(APPEND KMLBASE_DEPENDS ZLIB)
-  list(APPEND MINIZIP_DEPENDS ZLIB)
-endif()
-
-find_package(MiniZip)
-if(MINIZIP_FOUND)
-  include_directories(${MINIZIP_INCLUDE_DIR})
-else()
-  include(External_minizip)
-  list(APPEND KMLBASE_DEPENDS MINIZIP)
-endif()
-
-find_package(UriParser)
-if(URIPARSER_FOUND)
-  include_directories(${URIPARSER_INCLUDE_DIR})
-else()
-  include(External_uriparser)
-  list(APPEND KMLBASE_DEPENDS URIPARSER)
-endif()
-
-find_package(Boost)
-if(Boost_FOUND)
-  message(STATUS "Found Boost: ${Boost_VERSION}")
-else()
-  include(External_boost)
-  list(APPEND KMLBASE_DEPENDS BOOST)
-  message(STATUS "Found Boost includes: ${Boost_INCLUDE_DIRS}")
-endif()
-
-include_directories(${Boost_INCLUDE_DIRS})
+include_directories(${CONAN_INCLUDE_DIRS_BOOST})
 
 if(WITH_SWIG)
   find_package(SWIG)
@@ -149,10 +107,10 @@ if(WITH_SWIG)
   endif()
 endif()
 
-list(APPEND KMLBASE_LINK_LIBS ${EXPAT_LIBRARY})
-list(APPEND KMLBASE_LINK_LIBS ${ZLIB_LIBRARY})
-list(APPEND KMLBASE_LINK_LIBS ${MINIZIP_LIBRARY})
-list(APPEND KMLBASE_LINK_LIBS ${URIPARSER_LIBRARY})
+list(APPEND KMLBASE_LINK_LIBS ${CONAN_LIBS_EXPAT})
+list(APPEND KMLBASE_LINK_LIBS ${CONAN_LIBS_ZLIB})
+list(APPEND KMLBASE_LINK_LIBS ${CONAN_LIBS_MINIZIP})
+list(APPEND KMLBASE_LINK_LIBS ${CONAN_LIBS_URIPARSER})
 
 include_directories(${CMAKE_SOURCE_DIR}/src)
 
