--- a/src/kml/base/mimetypes.cc
+++ b/src/kml/base/mimetypes.cc
@@ -25,14 +25,17 @@
 
 // This defines some common KML-related mimetype strings.
 
+#ifdef WIN32
+#define MIMETYPE_DEFINES
+#endif
 #include "kml/base/mimetypes.h"
 
 namespace kmlbase {
 
-const char* kAtomMimeType = "application/atom+xml";
-const char* kCsvMimeType = "text/csv";
-const char* kKmlMimeType = "application/vnd.google-earth.kml+xml";
-const char* kKmzMimeType = "application/vnd.google-earth.kmz";
+MIMETYPE_DECL const char* kAtomMimeType = "application/atom+xml";
+MIMETYPE_DECL const char* kCsvMimeType = "text/csv";
+MIMETYPE_DECL const char* kKmlMimeType = "application/vnd.google-earth.kml+xml";
+MIMETYPE_DECL const char* kKmzMimeType = "application/vnd.google-earth.kmz";
 
 }
 
--- a/src/kml/base/mimetypes.h
+++ b/src/kml/base/mimetypes.h
@@ -28,12 +28,22 @@
 #ifndef KML_BASE_MIMETYPES_H__
 #define KML_BASE_MIMETYPES_H__
 
+#ifdef WIN32
+  #ifdef MIMETYPE_DEFINES
+    #define MIMETYPE_DECL __declspec(dllexport)
+  #else
+    #define MIMETYPE_DECL __declspec(dllimport)
+  #endif
+#else
+  #define MIMETYPE_DECL
+#endif
+
 namespace kmlbase {
 
-extern const char* kAtomMimeType;
-extern const char* kCsvMimeType;
-extern const char* kKmlMimeType;
-extern const char* kKmzMimeType;
+MIMETYPE_DECL extern const char* kAtomMimeType;
+MIMETYPE_DECL extern const char* kCsvMimeType;
+MIMETYPE_DECL extern const char* kKmlMimeType;
+MIMETYPE_DECL extern const char* kKmzMimeType;
 
 }  // end namespace kmlbase
 
